### DISCUSSION TITLE
Disable lsp-ui-doc on cursor by default

### DISCUSSION THREAD

**Discussion by ericdallo** [emacs-lsp/lsp-mode#3182] [Q&A]
I'm opening this as IMO we should not show the `lsp-ui-doc` glance popup when cursor stops on a symbol, IMO we should only show if hovering with the mouse, this causes a lot of notice on lsp-ui package.

WDYT?

**Comment by madscientist** [comment]
Can you elaborate on exactly how this would work?  As I mentioned in gitter yesterday I love being able to get docs, but as a focus-follows-mouse user it badly breaks my workflow because as I move point around if a doc popup appears under my mouse cursor it "breaks" continuous key presses.  For example, I use `C-s` then type something then start hitting `C-s` repeatedly; when the popup breaks this it stops repeating.  Continuously hitting `C-s` doesn't help because it constantly breaks.  I have to remember to keep my mouse cursor down near the bottom of the window to reduce the chances that the popup appears under it, which I consistently fail to do.

Even more galling is when I'm in the middle of an undo cycle: once the undo cycle is broken then it starts to redo!  I am so used to the traditional undo in Emacs, and normally like it, that I can't switch to one of the new different undo methods.

Anyway.  I would like it if I explicitly move the mouse to hover over something and then a doc popup appeared.  But how is that distinguished from, say, page down and your mouse now happens to hover over some symbol?  Would a doc popup appear here as well?

I've been meaning to investigate how `lsp-ui-doc-show-with-mouse` and `lsp-ui-doc-show-with-cursor` work but haven't found time: how is this request different from those?

**Reply by ericdallo** [reply]
Yeah, the point here is more about disabling the doc popup showing when you just stop the cursor in a symbol without any extra action, the mouse hover is more like a improvement IMO.
AFAIK there is no editors where show hover/documentation popups only when you move the cursor to a symbol, so it sounds noisy to me

**Reply by madscientist** [reply]
Well you can already disable the popup: just set `lsp-ui-doc-enable` to `nil` :).  The question is really, what operation should _show_ the popup and how exactly will it work (from a user perspective)?

Thanks for raising this.

**Reply by ericdallo** [reply]
I have my `lsp-ui-doc-enable` disabled already, the problem I see here is that when `lsp-ui-doc-enable` is `t`, it shows only with cursor hover without any user action like calling a command or hovering with mouse instead of cursor, which makes most users want to disable that annoying popup.
When I want to see the docs I can manually trigger `lsp-ui-doc-glance` or `lsp-ui-doc-show`.

**Reply by sebeaumont** [reply]
Ok if you use a mouse -- believe it or not there are some of us that don't -- also I've noticed that `lsp-haskell` is not reliable in showing the brief type info the modeline (which is odd as the doc seems to work better) thus so I've had to re-enable it.

**Reply by yyoncho** [reply]
> also I've noticed that lsp-haskell is not reliable in showing the brief type info the modeline (which is odd as the doc seems to work better) thus so I've had to re-enable it.

@sebeaumont 

Most likely setting lsp-eldoc-render-all to t will fix this issue.

**Reply by sebeaumont** [reply]
@yyoncho Thanks for the tip.

**Comment by Martinsos** [comment]
If I am correct, lsp-ui-doc on cursor is now disabled on default, is it? In that case, is there a way to re-enable this behaviour via some setting?

**Reply by ericdallo** [reply]
yes, `(setq lsp-ui-doc-show-with-cursor t)`
